library/Process-darwin.cpp

```diff
@@ -59,21 +59,17 @@ Process::Process(VersionInfoFactory * known_versions)
         real_path = realpath(path, NULL);
     }
 
-    identified = false;
-    my_descriptor = 0;
-    my_pe = 0;
-
     md5wrapper md5;
     uint32_t length;
     uint8_t first_kb [1024];
     memset(first_kb, 0, sizeof(first_kb));
     // get hash of the running DF process
     my_md5 = md5.getHashFromFile(real_path, length, (char *) first_kb);
     // create linux process, add it to the vector
-    VersionInfo * vinfo = known_versions->getVersionInfoByMD5(my_md5);
+    const VersionInfo * vinfo = known_versions->getVersionInfoByMD5(my_md5);
     if(vinfo)
     {
-        my_descriptor = new VersionInfo(*vinfo);
+        my_descriptor.reset(new VersionInfo(*vinfo));
```

Comment:
Maybe `my_description = std::move(dts::make_unique<VersionInfo>(*vinfo)`

Alternative would be making my_descriptor shared_ptr and just assigning returned `shared_ptr<VersionInfo>&` to my_descriptor.

Reply:
As mentioned in the other comment, I don't want to make too many changes to the code just yet. I did however use the make_unique call.